### PR TITLE
kernel: bump 6.6 to 6.6.39

### PR DESCRIPTION
**Pull request by graysky2**
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.39

Manually rebased:
	generic/backport-6.6/0080-v6.9-smp-Avoid-setup_max_cpus_namespace_collision_shadowing.patch

Removed upstreamed:
	generic/backport-6.6/801-v6.11-gpio-mmio-do-not-calculate-bgpio_bits-via-ngpios.patch[1]
	generic/backport-6.6/0081-v6.10-cpu-Fix-broken-cmdline-nosmp-and-maxcpus-0.patch[2]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.39&id=dee87316b5f5f167a201491a774bbd6e10c8dd94
2. https://github.com/gregkh/linux/commit/69787793e7f0673465c89714d1522fde978bbfa8

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/glinet_gl-mt6000, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/glinet_gl-mt6000, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

### CODE REVIEW

File: target/linux/generic/backport-6.6/0080-v6.9-smp-Avoid-setup_max_cpus_namespace_collision_shadowing.patch [hunk 1]
```diff
@@ -32,13 +32,16 @@ Cc: linux-kernel@vger.kernel.org
...
 +++ b/kernel/cpu.c
-@@ -1905,14 +1905,14 @@ static bool __init cpuhp_bringup_cpus_pa
+@@ -1905,17 +1905,17 @@ static bool __init cpuhp_bringup_cpus_pa
  static inline bool cpuhp_bringup_cpus_parallel(unsigned int ncpus) { return false; }
  #endif /* CONFIG_HOTPLUG_PARALLEL */
  
 -void __init bringup_nonboot_cpus(unsigned int setup_max_cpus)
 +void __init bringup_nonboot_cpus(unsigned int max_cpus)
  {
+ 	if (!setup_max_cpus)
```
Comment (hauke) [inline target/linux/generic/backport-6.6/0080-v6.9-smp-Avoid-setup_max_cpus_namespace_collision_shadowing.patch:42]:
```suggestion
 	if (!max_cpus)
```
The upstream version of this patch does it too: https://github.com/torvalds/linux/commit/6ef8eb5125722c241fd60d7b0c872d5c2e5dd4ca

File: target/linux/generic/backport-6.6/0081-v6.10-cpu-Fix-broken-cmdline-nosmp-and-maxcpus-0.patch
Comment (hauke) [inline target/linux/generic/backport-6.6/0081-v6.10-cpu-Fix-broken-cmdline-nosmp-and-maxcpus-0.patch:1]: This patch was applied upstream: https://github.com/gregkh/linux/commit/69787793e7f0673465c89714d1522fde978bbfa8
Please remove this patch

**Comment by graysky2** [comment]
@hauke - as you see, I commited your suggestion using github's web UI.  How can I pull this change into my fork so I can rebase/squash then remove the 2nd patch, etc.?

File: target/linux/generic/backport-6.6/0081-v6.10-cpu-Fix-broken-cmdline-nosmp-and-maxcpus-0.patch
Comment (graysky2) [reply target/linux/generic/backport-6.6/0081-v6.10-cpu-Fix-broken-cmdline-nosmp-and-maxcpus-0.patch:1]: Done

**Comment by graysky2** [comment]
@hauke - shall I leave this as two commits?

**Review by graysky2 — COMMENTED** [review]
@hauke - do you agree with this change?  If so I will squash/rebase.

**Comment by hauke** [comment]
> @hauke - as you see, I commited your suggestion using github's web UI. How can I pull this change into my fork so I can rebase/squash then remove the 2nd patch, etc.?
> 
> EDIT: I think I got it with:
> 
> ```
> git pull upstream pull/15928/head
> (edit patch)
> git merge --continue
> ```

You should set this as a remote, for example this should work `git branch --set-upstream-to=origin/6` Replace `origin` with the name of your repo.
Then you can just do `git pull`.

**Comment by hauke** [comment]
> @hauke - do you agree with this change? If so I will squash/rebase.

looks fine, just squash it all together.

**Comment by graysky2** [comment]
Done, I added your `Co-authored-by: Hauke Mehrtens <hauke...>` as you see :handshake:

File: target/linux/generic/backport-6.6/0080-v6.9-smp-Avoid-setup_max_cpus_namespace_collision_shadowing.patch (hunk 1 above)
Comment (DragonBluep) [reply target/linux/generic/backport-6.6/0080-v6.9-smp-Avoid-setup_max_cpus_namespace_collision_shadowing.patch]: > The upstream version of this patch does it too: [torvalds/linux@6ef8eb5](https://github.com/torvalds/linux/commit/6ef8eb5125722c241fd60d7b0c872d5c2e5dd4ca)

@graysky2 Please remove this upstreamed patch.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.39&id=69787793e7f0673465c89714d1522fde978bbfa8

**Pull request merged**